client/pkg/transport/listener.go

```diff
@@ -527,8 +528,28 @@ func (info TLSInfo) ServerConfig() (*tls.Config, error) {
 		cfg.ClientCAs = cp
 	}
 
-	// "h2" NextProtos is necessary for enabling HTTP2 for go's HTTP server
-	cfg.NextProtos = []string{"h2"}
+	// If manually defined ciphers are provided and TLS1.3 is not allowed, the
+	// list of ciphers may not allow HTTP2.  Test for this and disable if this is the case.
+	if len(cfg.CipherSuites) > 0 && cfg.MaxVersion > 0 && cfg.MaxVersion < tls.VersionTLS13 {
+
+		// Test for the cases where TLS1.3 is not allowed, and thus http2 may not
+		// be possible when select cipherSuites are defined.
+		var foundHTTP2Cipher bool
+		for _, c := range cfg.CipherSuites {
+			switch c {
+			// If one of these cipher suites are defined, then the h2 protocol is possible.
+			case tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256:
```

Comment:
Nit: There is a longstanding upstream golang issue and cl which proposes making these bad ciphers easier to test for so we can potentially simplify this in future once that is done.  Perhaps we should add a `// TODO:` in here mentioning the upstream issue so we can revisit in future: https://github.com/golang/go/issues/41068#issuecomment-722549561

Reply:
How about this... avoid TODO by letting the http2 ConfigureServer(s *http.Server, conf *Server) throw the built-in error when setting the cipher set.

The e2e test will see a different reply if this http2 cipher set changes, but a failing test can be adjusted when the test fails-- there's no real need to add a TODO in an e2e test as a failure will imply a TODO.

Reply:
I stand corrected; a TODO is needed, as you suggested.  My apologies.  Please let me know if you have any further considerations.